extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -561,9 +561,12 @@ export class AmpA4A extends AMP.BaseElement {
    */
   shouldInitializePromiseChain_() {
     const slotRect = this.getIntersectionElementLayoutBox();
-    if (
+    const fixedSizeZeroHeightOrWidth =
       this.getLayout() != Layout.FLUID &&
-      (slotRect.height == 0 || slotRect.width == 0)
+      (slotRect.height == 0 || slotRect.width == 0);
+    if (
+      fixedSizeZeroHeightOrWidth ||
+      this.element.classList.contains('i-amphtml-hidden-by-media-query')
```

Comment:
/cc @dvoytenko to take a look

Reply:
So, basically, what you're looking for here is whether the element is actually `display:none`. Right?

Reply:
I think so. Unless `i-amphtml-hidden-by-media-query` does something else under the hood to keep the element hidden.

Reply:
Hmm. My questions then are:

1. Do we currently set `hidden` attribute for `i-amphtml-hidden-by-media-query` case as well?
2. Can we check it's actually indeed `display:none`?

Depending on these answers, we'd likely be better of using the `hidden` attribute if it's indeed set, or alternatively checking `getComputedStyle(x).display == 'none'`.

Reply:
I do not see a `hidden` attribute on slots hidden by media query. Also, I'm not sure why getComputedStyle is the better option, since it's much more expensive than iterating through a list of classes on an element (unless I'm missing something?). If I'm reading the code right (https://github.com/ampproject/amphtml/blob/master/src/custom-element.js#L691), `i-amphtml-hidden-by-media-query` is set precisely on those slots which we don't want to issue a request, so checking for its inclusion seems like a reasonable thing to do.

Reply:
Ok. Let's do this:

1. Let's check both: `element.classList.contains('i-amphtml-hidden-by-media-query') || element.hasAttribute('hidden')`.
2. Let's leave a `TODO` that more `display:none` case might need to be supported.

Reply:
Done!